### DISCUSSION TITLE
hope to obtain sourceFile and sourceLine through the Ghidra API, similar to idaapi.get_source_linnum() and idaapi.get_sourcefile().

### DISCUSSION THREAD

**Discussion by funny-mud-peee** [NationalSecurityAgency/ghidra#7246] [Q&A] [answered]
I want to know how to obtain the source file and line number through the Ghidra API, similar to how IDA provides interfaces: idaapi.get_source_linnum() and idaapi.get_sourcefile().

**Comment by ghidracadabra** [comment, marked as answer]
We've recently added this capability.  It's in the master branch but not yet in an official release. You can get this information from a program's `SourceFileManager` (`currentProgram.getSourceFileManager()` in a script).  The DWARF and PDB analyzers have been updated to record this information by default.

**Reply by dev747368** [reply]
> We've recently added this capability. It's in the master branch but not yet in an official release. You can get this information from a program's `SourceFileManager` (`currentProgram.getSourceFileManager()` in a script). The DWARF and PDB analyzers have been updated to record this information by default.

and go!
